pointcloud/basic_octree.go

```diff
@@ -182,3 +214,55 @@ func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, thres
 	}
 	return false, errors.New("unknown octree node type")
 }
+
+// CollidesWith checks if the given octree collides with the given geometry and returns true if it does.
+func (octree *BasicOctree) CollidesWith(geom spatialmath.Geometry) (bool, error) {
```

Comment:
I believe the implementation of `CollidesWithGeometry` currently checks for a collision against a box constructed of the entire occupied octree octant, rather than the individual observed point it contains. This should be corrected for this PR

Reply:
It looks like the point is checked for collision when the node is `leafNodeFilled`. an `internalNode` doesn't contain any point. I don't understand what the required change is.

Reply:
Sorry- you're correct. Disregard this.